src/main/java/io/supertokens/inmemorydb/queries/OAuthQueries.java

```diff
@@ -212,51 +230,50 @@ public static boolean deleteOAuthClient(Start start, String clientId, AppIdentif
-                + "(app_id, target_type, target_value, timestamp, exp) VALUES (?, ?, ?, ?, ?) "
-                + "ON CONFLICT (app_id, target_type, target_value) DO UPDATE SET timestamp = ?, exp = ?";
+        String DELETE = "DELETE FROM " + Config.getConfig(start).getOAuthSessionsTable()
+                        + " WHERE gid = ? and app_id = ?";
+        int numberOfRows = update(start, DELETE, pst -> {
+            pst.setString(1, gid);
+            pst.setString(2, appIdentifier.getAppId());
+        });
+        return numberOfRows > 0;
+    }
 
-        long currentTime = System.currentTimeMillis() / 1000;
-        update(start, INSERT, pst -> {
+    public static boolean deleteOAuthSessionByClientId(Start start, AppIdentifier appIdentifier, String clientId)
+            throws SQLException, StorageQueryException {
+        String DELETE = "DELETE FROM " + Config.getConfig(start).getOAuthSessionsTable()
+                + " WHERE app_id = ? and client_id = ?";
+        int numberOfRows = update(start, DELETE, pst -> {
             pst.setString(1, appIdentifier.getAppId());
-            pst.setString(2, targetType.getValue());
-            pst.setString(3, targetValue);
-            pst.setLong(4, currentTime);
-            pst.setLong(5, exp);
-            pst.setLong(6, currentTime);
-            pst.setLong(7, exp);
+            pst.setString(2, clientId);
         });
+        return numberOfRows > 0;
     }
 
-    public static boolean isOAuthTokenRevokedBasedOnTargetFields(Start start, AppIdentifier appIdentifier, OAuthRevokeTargetType[] targetTypes, String[] targetValues, long issuedAt)
+    public static boolean deleteOAuthSessionBySessionHandle(Start start, AppIdentifier appIdentifier, String sessionHandle)
             throws SQLException, StorageQueryException {
-        String QUERY = "SELECT app_id FROM " + Config.getConfig(start).getOAuthRevokeTable() +
-                " WHERE app_id = ? AND timestamp >= ? AND (";
-
-        for (int i = 0; i < targetTypes.length; i++) {
-            QUERY += "(target_type = ? AND target_value = ?)";
+        String DELETE = "DELETE FROM " + Config.getConfig(start).getOAuthSessionsTable()
+                + " WHERE app_id = ? and session_handle = ?";
+        int numberOfRows = update(start, DELETE, pst -> {
+            pst.setString(1, appIdentifier.getAppId());
+            pst.setString(2, sessionHandle);
+        });
+        return numberOfRows > 0;
+    }
 
-            if (i < targetTypes.length - 1) {
-                QUERY += " OR ";
-            }
+    public static boolean deleteJTIFromOAuthSession(Start start, AppIdentifier appIdentifier, String gid, String jti)
+            throws SQLException, StorageQueryException {
+        //jti is a comma separated list. When deleting a jti, just have to delete from the list
+        List<String> savedJTIs  = getOAuthJTIsByGID(start, appIdentifier, gid);
```

Comment:
Could we do this in a single op by just string replacing it?

Reply:
done